web/src/main.tsx

```diff
@@ -63,6 +73,116 @@ const queryClient = new QueryClient({
 
 refetchDataOnHourChange(queryClient);
 
+export function ValidZoneIdGuardWrapper({ children }: { children: JSX.Element }) {
+  const [searchParameters] = useSearchParams();
+  const { zoneId, timeAverage } = useParams();
+  if (!zoneId) {
+    return <Navigate to="/map/hourly" replace />;
+  }
+  const upperCaseZoneId = zoneId.toUpperCase();
+  if (zoneId !== upperCaseZoneId) {
+    return (
+      <Navigate
+        to={`/zone/${upperCaseZoneId}/${timeAverage}?${searchParameters}`}
+        replace
+      />
+    );
+  }
+
+  // Handle legacy Australia zone names
+  if (upperCaseZoneId.startsWith('AUS')) {
+    return (
+      <Navigate
+        to={`/zone/${zoneId.replace('AUS', 'AU')}/${timeAverage}?${searchParameters}`}
+        replace
+      />
+    );
+  }
+
+  // Only allow valid zone ids
+  // TODO: This should redirect to a 404 page specifically for zones
+  if (!zoneExists(upperCaseZoneId)) {
+    return <Navigate to="/map/hourly" replace />;
+  }
+
+  return children;
+}
+
+function HandleLegacyRoutes() {
+  const [searchParameters] = useSearchParams();
+
+  const page = (searchParameters.get('page') || 'map')
+    .replace('country', 'zone')
+    .replace('highscore', 'ranking');
+  searchParameters.delete('page');
+
+  const zoneId = searchParameters.get('countryCode');
+  searchParameters.delete('countryCode');
+
+  return (
+    <Navigate
+      to={{
+        pathname: zoneId ? `/zone/${zoneId}` : `/${page}`,
+        search: searchParameters.toString(),
+      }}
+    />
+  );
+}
+const router = createBrowserRouter([
+  {
+    path: '/',
+    element: <App />,
+    children: [
+      {
+        path: '/',
+        element: <HandleLegacyRoutes />,
+      },
+      {
+        path: '/map',
+        element: (
+          <Navigate
+            to="/map/hourly"
+            replace
+            state={{ preserveSearch: true, preserveHash: true }}
+          />
+        ),
+      },
+      {
+        path: '/zone',
+        element: (
+          <Navigate
+            to="/map/hourly"
+            replace
+            state={{ preserveSearch: true, preserveHash: true }}
+          />
+        ),
+      },
+      {
+        path: '/map/:urlTimeAverage?/:urlDatetime?',
+        element: <RankingPanel />,
+      },
+      {
+        path: '/zone/:zoneId/:urlTimeAverage?/:urlDatetime?',
+        element: (
+          <ValidZoneIdGuardWrapper>
+            <Suspense fallback={<LoadingSpinner />}>
+              <ZoneDetails />
+            </Suspense>
+          </ValidZoneIdGuardWrapper>
+        ),
+      },
+      {
+        path: '*',
```

Comment:
nit: We probably want to a not found page or something here (but I understand that this may be out of scope)

Reply:
I have a draft PR to add it for zones and will do a general one as well once this is merged.

It will need to be adapted to the new router location.